<launch>

    <param name="uav_model" command="$(find xacro)/xacro $(find uav_swarm_description)/urdf/quadrotor.urdf.xacro" />

    <arg name="pluginlists_yaml" default="$(find mavros)/launch/apm_pluginlists.yaml" />
    <arg name="config_yaml" default="$(find uav_swarm_control)/launch/apm_config.yaml" />

    <arg name="origin_lat" default="-22.413949" />
    <arg name="origin_lon" default="-45.446555" />

    <group ns="uav0">
            <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                    <param name="fcu_url" value="udp://127.0.0.1:14551@" />
                    <param name="gcs_url" value="" />
                    <param name="target_system_id" value="1" />
                    <param name="target_component_id" value="1" />

                    <!-- load blacklist, config -->
                    <rosparam command="load" file="$(arg pluginlists_yaml)" />
                    <rosparam command="load" file="$(arg config_yaml)" />
            </node>

            <node pkg="uav_swarm_control" type="gas_finder_node" name="gas_finder_node" output="screen" >
                <param name="uav_id" value="0" />
                <param name="origin_lat" value="$(arg origin_lat)" />
                <param name="origin_lon" value="$(arg origin_lon)" />
            </node>

            <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" >
                <param name="uav_id" value="0" />
            </node>
    </group>

    <group ns="uav1">
            <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                    <param name="fcu_url" value="udp://127.0.0.1:14561@" />
                    <param name="gcs_url" value="" />
                    <param name="target_system_id" value="1" />
                    <param name="target_component_id" value="1" />

                    <!-- load blacklist, config -->
                    <rosparam command="load" file="$(arg pluginlists_yaml)" />
                    <rosparam command="load" file="$(arg config_yaml)" />
            </node>

            <node pkg="uav_swarm_control" type="gas_finder_node" name="gas_finder_node" output="screen" >
                <param name="uav_id" value="1" />
                <param name="origin_lat" value="$(arg origin_lat)" />
                <param name="origin_lon" value="$(arg origin_lon)" />
            </node>

            <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" >
                <param name="uav_id" value="1" />
            </node>
    </group>

    <group ns="uav2">
            <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" >
                    <param name="fcu_url" value="udp://127.0.0.1:14571@" />
                    <param name="gcs_url" value="" />
                    <param name="target_system_id" value="1" />
                    <param name="target_component_id" value="1" />

                    <!-- load blacklist, config -->
                    <rosparam command="load" file="$(arg pluginlists_yaml)" />
                    <rosparam command="load" file="$(arg config_yaml)" />
            </node>

            <node pkg="uav_swarm_control" type="gas_finder_node" name="gas_finder_node" output="screen" >
                <param name="uav_id" value="2" />
                <param name="origin_lat" value="$(arg origin_lat)" />
                <param name="origin_lon" value="$(arg origin_lon)" />
            </node>

            <node pkg="uav_swarm_control" type="marker_publisher" name="marker_publisher" >
                <param name="uav_id" value="1" />
            </node>
    </group>

    <!-- Run RVIZ-->
    <node name="$(anon rviz)" pkg="rviz" type="rviz" args="$(find uav_swarm_control)/uav_swarm_control.rviz" />

</launch>
